### PR TITLE
PS3 Chris Erickson

### PR DESCRIPTION
**Pull request by christopherickson**
Hi, so I'm not done (sorry, on the R studio short bus), but close. I have some quesitons. A few just so that I understand, and some that are holding back homework completion (#4 is critical if you don't have time to answer all)

1.	What are 'key' and 'value' for the gather( ), and separate( ) verbs (can't really understand from the descriptions?
2.	Why so we need to change 6hpi to hpi.6?
3.	What exactly is spread doing? Looks like its taking 1 column and making it 4, in this case
**4.**	Fatal error: I get an error line 175 that cleaned_data_2 does not exist. Not sure why, because it works if I run line by line, but it does not knit. 
5.	What exactly is group_by doing? 
**6.**	I get lost at the end of #3. How to select for the highest mean, and lowest variance at the simultaneously? Maybe I need to brush up on my stats?
7.	For #4 when we use the in line scripting, can we just insert r script between text? I can't tell cause not knitting (#4)
8.	Why do we have to remove gene.name when using slice (#4 tip under question 3)?

### CODE REVIEW

**Comment by mandylr** [comment]
1. the key and value for gather/separate are sort of just column names you supply. In the case of gather(), the "key" would be what you want the **new** column name to be for what is currently the "messy" column names. The "value" would be measurement of whatever is within those columns. See [here](https://blog.rstudio.org/2014/07/22/introducing-tidyr/) for actual examples. 

2. You don't *need* to change the 6hpi to hpi.6, but it makes it easier in the long run. It's common practice for these types of variable to put the common part (e.g. hpi) in the front and the value last. The hpi.6 format also makes it easy if you want to separate just the numerical values from the columns. Essentially, it just makes data manipulation easier. 

3. spread is the opposite of gather. If you have 2 columns for instance drug and heart rate. Where the drug column contains names of 2 drugs, and the heart rate column contains the heart rate of people on those drugs. If you were to spread this data, you would now have 2 columns (for drug a and b), and the values within these columns would be heart rate of people on said drugs. In this case, if you spread the expression data and the time point, then you get 4 columns corresponding to the 4 time points (hpi.6, etc). and the values within those columns are the expression values. 

4. ? Fatal error. It won't knit because in the previous code block (reformat_time), line 147 you have a ```eval = FALSE``` tag, which means when you knit, that code block won't run. The reformat_time code block is where you define the cleaned_data_2 variable. Remove the ```eval = FALSE``` and it should work fine. 

other 4. ```group_by()``` alone defines groups in the data. So if you want to make a group of plant and fungus, then R will make groups of unique plant and fungus combinations. Then, when you use functions like ```summarize()```R will summarize within each group, rather than summarizing the whole data set. So for every unique pair of plant and fungus, it will calculate the summary statistic for only that group. If you group_by(plant), then it will create groups of unique plants (ignoring the rest of the data). When you summarize, R will calculate summary statistics for each plant. 

6. This is a tricky one, and not super important to understanding R (or stats). What you can do is ```arrange()``` . If you include 2 different variables to arrange by, it will order the first one listed (e.g. mean), then break any ties by ordering the second variable listed (e.g. variance). This doesn't work very well with 2 numeric variables because it's rare to get ties when calculating the mean and variance for your data. For the purposes of this question, just arrange by mean and variance (order of your choice), and report what you get. 

5. Yep. Look [here](https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf) for how to format. When the document knits, it evaluates the code and the output is the result of the code. 

6. ```slice()``` adheres to groupings. So if you group by plant, fungus, AND gene.name, then when you slice,  you will get the top results for this grouping (this won't give you the top 3 gene.names because they're included in the groups).  You need to group by only plant and fungus, then slice the top 3 gene.names based on the summary stats for those groups. 

6. ```slice()``` adheres to groupings. So if you group by plant, fungus, AND gene.name, then when you slice,  you will get the top results for this grouping (this won't give you the top 3 gene.names because they're included in the groups).  You need to group by only plant and fungus, then slice the top 3 gene.names based on the summary stats. 

Hope this helps!

**Comment by christopherickson** [comment]
I pushed PS3 with some revisions. Let me know if you don't get it, not sure if it made the pull request? Done with everything except #5, nearly done with that

**Comment by christopherickson** [comment]
I pushed the html file. Can't get the smooth line to show, but everything else looks fine. Let me know if you don't get it, not sure if it made the pull request?

File: README.Rmd [hunk 1]
```diff
@@ -138,12 +170,33 @@ cleaned_data %>%
...
+
 ```
 
 ## Problem 4
 
 Now examine the above final data frame above and write a few sentences putting inline code in least 3 places. For example, There are `r nrow(mtcars)` rows of data in `mtcars`.
 
+This analysis in problem 3 finds the highest difference in gene expression across the time points from 6 to 24hrs with the lowest variance. 
+
+START: `r nrow(raw_data)` genes were analyzed in `r length(unique(cleaned_data$fungus))*length(unique(cleaned_data$plant))` plant-fungi combos. Of these genes, the top 3 most consistently differentially expressed were `r ungroup(top3) %>% select(gene.name) %>% slice(1)`, `r ungroup(top3) %>% select(gene.name) %>% slice(2)`, and `r ungroup(top3) %>% select(gene.name) %>% slice(3)`.
```
Comment (mandylr) [inline README.Rmd]: Nice use of inline code

File: README.Rmd [hunk 2]
```diff
@@ -159,4 +212,11 @@ Plot the expression (`value`) by time (`hpi`) for the above genes. Format the pl
 Extra credit: add error bars to the plot (use `geom_errorbar`).
 
 ```{r problem_5}
+
+filter(cleaned_data, gene.name == "bgh02634" | gene.name == "bgh06450" | gene.name == "bghG000349000001001") -> plotme
```
Comment (mandylr) [inline README.Rmd]: This is not the same code as the html document, but the html code looks good!

**Comment by christopherickson** [comment]
Still can't see the smooth line. Changed the time.points to numerical value, maybe need another argument in geom_smooth? but nothing I've tried will add the line

**Pull request closed without merging**